gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DocumentReferenceMapper.java

```diff
@@ -0,0 +1,218 @@
+
+        return documentReference;
+    }
+
+    private List<RCMRMT030101UK04NarrativeStatement> getNarrativeStatements(RCMRMT030101UK04EhrComposition ehrComposition) {
+        return ehrComposition.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component4::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
+
+    private CodeableConcept getType(RCMRMT030101UK04NarrativeStatement narrativeStatement) {
+        var referenceToExternalDocument = narrativeStatement.getReference().get(0).getReferredToExternalDocument();
+
+        if (referenceToExternalDocument != null && referenceToExternalDocument.getCode() != null) {
+            if (referenceToExternalDocument.getCode().getOriginalText() == null
+                && referenceToExternalDocument.getCode().getDisplayName() != null) {
+                return codeableConceptMapper.mapToCodeableConcept(referenceToExternalDocument.getCode())
+                    .setText(referenceToExternalDocument.getCode().getDisplayName());
+            } else if (referenceToExternalDocument.getCode().getOriginalText() != null) {
+                return codeableConceptMapper.mapToCodeableConcept(referenceToExternalDocument.getCode())
+                    .setText(referenceToExternalDocument.getCode().getOriginalText());
+            }
+        }
+
+        return null;
+    }
+
+    private InstantType getIndexed(RCMRMT030101UK04EhrExtract ehrExtract) {
+        if (ehrExtract.getAuthor() != null) {
+            if (ehrExtract.getAuthor().getTime() != null && ehrExtract.getAuthor().getNullFlavor() == null) {
+                return DateFormatUtil.parseToInstantType(ehrExtract.getAuthor().getTime().getValue());
+            }
+
+            return DateFormatUtil.parseToInstantType(ehrExtract.getAvailabilityTime().getValue());
+        }
+
+        return null;
+    }
+
+    private List<Reference> getAuthor(RCMRMT030101UK04NarrativeStatement narrativeStatement,
+        RCMRMT030101UK04EhrComposition ehrComposition) {
+        return List.of(ParticipantReferenceUtil.getParticipantReference(narrativeStatement.getParticipant(), ehrComposition));
+    }
+
+    private Reference createOrganizationReference(Organization organization) {
+        return new Reference(organization.getIdElement());
+    }
+
+    private String buildDescription(RCMRMT030101UK04NarrativeStatement narrativeStatement) {
+        var referenceToExternalDocumentText = narrativeStatement.getReference().get(0).getReferredToExternalDocument().getText();
+        if (isAbsentAttachment(narrativeStatement)) {
+            return PLACEHOLDER_VALUE;
+        }
+        if (referenceToExternalDocumentText != null) {
+            String filename = referenceToExternalDocumentText.getReference().getValue();
+            if (filename != null) {
+                return buildFileName(filename);
+            }
```

Comment:
if the fileName is null in this instance, what happens to the filename?

Reply:
> if the fileName is null in this instance, what happens to the filename?

I believe we just leave this blank if it has no file name as im not sure if there will always be a value for referreedToExternalDocument/Text/Reference/@value

Reply:
[Documentation for the text object](https://data.developer.nhs.uk/dms/mim/4.2.00/Data%20Types/DataTypes.htm#ED)

You're right, reference is 0:1. Find it strange theres no fallback here, maybe Dawid might know?

Reply:
According to XSD  "referredToExternalDocument/text" is mandatory. I'd keep the defensive approach though and check for null.